node_modules/oae-principals/lib/library.js

```diff
@@ -74,3 +85,86 @@ LibraryAPI.Search.registerLibrarySearch('memberships-library', ['group'], {
         }
     }
 });
+
+/**
+ * Get all the group memberships for a principal from the AuthzAPI. This will take into
+ * consideration group deletes that have happened in the system. Not only must deleted groups not
+ * show in the library, indirect group membership that has been broken by a deleted group must not
+ * show, which is the rationale of using the memberships graph
+ *
+ * @param  {String}     principalId             The id of the principal whose memberships to get
+ * @param  {Function}   callback                Standard callback function
+ * @param  {Object}     callback.err            An error that occurred, if any
+ * @param  {String[]}   callback.memberships    The ids of the groups of which the principal is directly or indirectly a member
+ * @api private
+ */
+var _getAllGroupMembershipsFromAuthz = function(principalId, callback) {
+    // Get the full memberships graph from the AuthzAPI. Note that this includes all groups
+    // including those that have since been marked as deleted
+    AuthzAPI.getPrincipalMembershipsGraph(principalId, function(err, graph) {
+        if (err) {
+            return callback(err);
+        }
+
+        // Extract all group ids from the graph, excluding the principal id we actually searched for
+        var allGroupIds =
+            _.chain(graph.getNodes())
+                .pluck('id')
+                .without(principalId)
+                .value();
+
+        // Determine which of the groups in the memberships graph have been deleted
+        AuthzDelete.isDeleted(allGroupIds, function(err, deleted) {
+            if (err) {
+                return callback(err);
+            }
+
+            // Delete all group nodes from the graph that have been deleted
+            _.chain(deleted)
+                .keys()
+                .each(function(deletedGroupId) {
+                    graph.removeNode(deletedGroupId);
+                });
+
+            // The resulting membership will be all group nodes that are reachable VIA a full
+            // outbound edge traversal ("member of") starting from the `principalId`
+            var memberships =
+                _.chain(graph.traverseOut(principalId))
+                    .pluck('id')
+                    .without(principalId)
+                    .value();
+
+            // Return the final memberships list, ordered in string id order
```

Comment:
It doesn't, and I don't think it even will even be guaranteed to be anymore in this case. When I first did this, I was under the impression that the `pageResources` thing would still page through things, then I realized I could just return all of them and force a `null` nextToken, so ordering wasn't really necessary anymore after I already wrote some of this